sklearn/tree/_criterion.pyx

```diff
@@ -291,6 +326,18 @@ cdef class ClassificationCriterion(Criterion):
         self.weighted_n_samples = weighted_n_samples
         self.weighted_n_node_samples = 0.0
```

Comment:
This is defined in `pointer_reset`?

Reply:
oh great point! One question I have is could `self.weighted_n_samples` still be useful if someone were to override the class?

Reply:
Yes because `self.weighted_n_samples` is used as the total weight which normalizes the weight in the node (`self.weighted_n_node_samples`). 

However, if you do ctrl+f in both criterion and splitter and tree, you'll see that the value does not change. Which makes sense. You pass in data (could be X, y, or both), and `sample_weight`. Sample weight is a vector of weights applied to each sample/row. `weighted_n_node_samples` though does because for each node you're considering different samples.

Reply:
Thank you for the explanation. This is super helpful!